File: Homework/publication.py
import datetime

class Publication:
    def __init__(self, companie: str, value: float, drop: float, variation: float, date: datetime.date):
        self.companie = companie
        self.value = value
        self.drop = drop
        self.variation = variation
        self.date = date

    def to_json(self):
        return {
            'companie': self.companie,
            'value': self.value,
            'drop': self.drop,
            'variation': self.variation,
            'date': self.date
        }
    
    @staticmethod
    def from_json(json):
        return Publication(
            json['companie'],
            json['value'],
            json['drop'],
            json['variation'],
            datetime.datetime.strptime(json['date'], '%d-%m-%Y').date()
        )

File: POC_RabbitMq/src/main/java/ebs/generator/entities/Publication.java
package ebs.generator.entities;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.json.JSONObject;

import java.sql.Date;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Publication {
    private String company;
    private Float value;
    private Float drop;
    private Float variation;
    private Date date;

    public JSONObject toJson(){
        return new JSONObject() {{
            put("company", company);
            put("value", value);
            put("drop", drop);
            put("variation", variation);
            put("date", date);
        }};
    }

    public static Publication fromJson(JSONObject jsonObject) {
        String companyValue = jsonObject.getString("company");
        Float valueValue = (float) jsonObject.getDouble("value");
        Float dropValue = (float) jsonObject.getDouble("drop");
        Float variationValue = (float) jsonObject.getDouble("variation");
        Date dateValue = Date.valueOf(jsonObject.getString("date"));

        return new Publication(companyValue, valueValue, dropValue, variationValue, dateValue);
    }

    @Override
    public String toString() {
        return toJson().toString(2);
    }
}
